### DISCUSSION TITLE
blog/2022/220602-ipv6-bypass-billing-system/

### DISCUSSION THREAD

**Discussion by giscus** [zhanbao2000/blog#15] [Announcements]
# blog/2022/220602-ipv6-bypass-billing-system/

在某些校园网计费系统中，IPv6 地址是不计入校园网流量的。这里提供了一种方法，可以通过架设 IPv6 代理绕过校园网计费系统，从而实现免流上网。

https://blog.arisa.moe/blog/2022/220602-ipv6-bypass-billing-system/

**Comment by Biiddd** [comment]
同在皇城某高校，计费策略相同，正在研究通过v6回家，然后通过家里的服务器进行转发，可否使用v2在家搭建server端，然后校园网内直接用v2全局代理

**Reply by zhanbao2000** [reply]
理论上只要有一个公网IPv6地址就可以。在家弄的话可能要注意一下你家里ISP分配的IP地址可能会是动态的，做个DDNS就行

**Comment by csekon** [comment]
写的不错，但我们学校不限）

**Comment by DE3PDARKFANTASY** [comment]
校友大佬你好，xdx求教下，我在获取ipv6时就遇到了一些问题。首先，一号床直插电脑访问v6test 9/10，只有dns未通过，完全正常。但是接入路由器后，路由器可以wan6可以获取到v6，地址查询如下
root@ImmortalWrt:~# ip -6 addr show dev eth1
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 state UP qlen 1000
    inet6 2001:da8:a012:2d4:ca9c:bbff:feb2:57b8/64 scope global dynamic noprefixroute 
       valid_lft 2591998sec preferred_lft 604798sec
    inet6 2001:da8:a012:2d4::30/128 scope global dynamic noprefixroute 
       valid_lft 604111sec preferred_lft 604111sec
    inet6 fdd8:4bef:4338:10::1/64 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::ca9c:bbff:feb2:57b8/64 scope link 
       valid_lft forever preferred_lft forever
root@ImmortalWrt:~# ip -6 route show | grep default
default from 2001:da8:a012:2d4::30 via fe80::86d9:31ff:fe80:cddc dev eth1  metric 512 
default from 2001:da8:a012:2d4::/64 via fe80::86d9:31ff:fe80:cddc dev eth1  metric 512 
但是在路由器终端中ping不通外网v6地址，第三跳就丢包了
root@ImmortalWrt:~# traceroute6 -n -s 2001:da8:a012:2d4::30 2001:4860:4860::8888
traceroute to 2001:4860:4860::8888 (2001:4860:4860::8888) from 2001:da8:a012:2d4::30, 30 hops max, 72 byte packets
 1  2001:da8:a012:2d4::1  0.944 ms  1.135 ms  0.782 ms
 2  2001:da8:a012:ff03::1  0.642 ms  0.552 ms  0.526 ms
 3  *  *  *
 4  *  *  *
 5  *
ai问了一大圈都没能解决，求教下大佬之前有没有遇到过类似情况，有什么解决方案吗。跪谢大佬了
坐标：东丽校区北院

**Reply by zhanbao2000** [reply]
之前我一直都是终端设备直接连到校园网，或者是把路由器当AP用，没试过让路由器来获取IPv6，所以也没遇到过类似情况。

看起来有点像是分配了IPv6但是实际上并没有登录（我不太记得在实际登录校园网之前是否会分配IPv6地址了）

**Reply by DE3PDARKFANTASY** [reply]
实际登陆前就会分配IP了，v4v6IP都有，电脑连上WiFi不管访问什么都会跳转到100.200认证登录，登录完了v4就正常访问了，登录前后访问这个http://[2001:da8:a012:ff06::252]:9002/都不通。谢谢大佬了，我去看看当ap能不能实现我们宿舍需求吧，舍友要打瓦低延时加免流下载，我要公网IPv6玩pt

**Reply by zhanbao2000** [reply]
免流和低延时是冲突的需求，毕竟免流走的是代理。

至于IPv6版本的登录页，访问不通可能是IP地址已经变了，毕竟离写这篇blog也过了那么多年了。不如试试用4.3.2的方法看看能不能复现获取现在版本的IPv6登录页

**Reply by DE3PDARKFANTASY** [reply]
谢谢大佬，改成ap果然好了，就是上行大水管变成百兆左右小水管了，可能是我路由器的问题吧

**Comment by blissky** [comment]
航专这ipv6时有时无，没学长的博客真不知道这是怎么认证的，写了个自动认证脚本扔openwrt上了，本来以为不用操心，结果航专一放假就给ipv6掐了
